server/src/document/document.service.ts

```diff
@@ -119,11 +138,30 @@ export class DocumentService {
           )
         `);
 
-        // If the record GUID also returns no Artifact, then there may be
-        // something wrong with the GUID 
         if (!firstArtifact) {
           const errorMessage = `No document access.`;
-          console.log(`Client attempted to retrieve document ${pathSegment}, but no associated public, submitted packages or artifacts were found.`);
+          console.log(`Client attempted to retrieve document ${path}, but no associated public, submitted artifacts were found.`);
+
+          throw new HttpException({
+            "code": "NO_DOCUMENT_ACCESS",
+            "detail": errorMessage,
+          }, HttpStatus.NOT_FOUND);
+        }
+      }
+
+      // similarly secure project action documents
+      if (entityType === "dcp_projectaction") {
```

Comment:
I mean, I guess this is fine, but I don't understand why we keep adding more `if`s into a method, making it longer, making it harder to understand. Make more methods. Split stuff out. Etc.

Reply:
I know the code is sloppy almost everywhere else, but we shouldn't add to it.

Reply:
https://sandimetz.com/99bottles

Reply:
@allthesignals  I added a refactor commit that sets up separate GET endpoints for each of the entities. Can you take a look and let me know your thoughts?